### PR TITLE
refactor: enforce name of the one-and-only time column

### PR DESCRIPTION
**Pull request by crepererum**
We currently only support a single time dimension and some parts of other stack rely on the name of the time column. So lets enforce the name (note that `schema::try_from_arrow` already checks for duplicate column, so we are now left with a single dimension).

### CODE REVIEW

File: schema/src/lib.rs [hunk 1]
```diff
@@ -89,6 +89,13 @@ pub enum Error {
         column_name: String,
         md: Option<String>,
     },
+
+    #[snafu(display(
+        "Time column should be named '{}' but is named '{}'",
```
Comment (alamb) [inline schema/src/lib.rs:94]:
```suggestion
        "Internal Error: Time column should be named '{}' but is named '{}'",
```

Comment (alamb) [reply schema/src/lib.rs:94]: maybe it is worth a hint to users that there is no way they are going to work around this lol

**Review by alamb — APPROVED** [review]
THERE CAN BE ONLY ONE

File: schema/src/lib.rs (hunk 1 above)
Comment (crepererum) [reply schema/src/lib.rs]: done

**Pull request merged**